From f3bb28bfa0e4574fccb8e5c54ee1bd1b1bdcac21 Mon Sep 17 00:00:00 2001
From: Dan Liang <dan.liang@intel.com>
Date: Thu, 2 May 2013 16:52:17 +0800
Subject: [PORT FROM MAIN] MatroskaExtractor: check the return value to avoid error propagation

BZ: 83053

pTP could be NULL and cause crash.

Change-Id: I783d6485acba363557e41faf81a9cd186db87a4e
Orig-Change-Id: Ifcfd803b2dd312807b527ca9caacbc2940f93a19
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 .../libstagefright/matroska/MatroskaExtractor.cpp  |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/matroska/MatroskaExtractor.cpp b/media/libstagefright/matroska/MatroskaExtractor.cpp
index 535ad3c..e4e92b7 100644
--- a/media/libstagefright/matroska/MatroskaExtractor.cpp
+++ b/media/libstagefright/matroska/MatroskaExtractor.cpp
@@ -386,7 +386,11 @@ void BlockIterator::seek(
     // Always *search* based on the video track, but finalize based on mTrackNum
     const mkvparser::CuePoint::TrackPosition* pTP;
     if (pTrack && pTrack->GetType() == 1) {
-        pCues->Find(seekTimeNs, pTrack, pCP, pTP);
+        bool ret = pCues->Find(seekTimeNs, pTrack, pCP, pTP);
+        if (!ret) {
+            ALOGE("Could not find the matching cue point");
+            return;
+        }
     } else {
         ALOGE("Did not locate the video track for seeking");
         return;
-- 
1.7.4.1

